src/st/no3/3FF00.c

```diff
@@ -0,0 +1,1208 @@
+#include "no3.h"
+
+void EntityRoomTransition2(Entity* self) {
+    Entity* newEntity;
+    Primitive* prim;
+    // Used as both a loop variable and a temp
+    s32 localVar;
+
+    // These are weird fake vars needed for matching
+    Entity* gents = g_Entities;
+    Unkstruct8* layout = &g_CurrentRoomTileLayout;
+
+    if (self->ext.generic.unk80.modeS32 == 0 && self->step < 4) {
```

Comment:
We are trying to phase out usage of `ext.generic`. Would you mind making a struct? See this PR https://github.com/Xeeynamo/sotn-decomp/pull/409